File: mjcbsn22/calculating_with_functions.py
def zero(f=None):
  return f[0](0, f[1]) if f else 0
def one(f=None):
  return f[0](1, f[1]) if f else 1
def two(f=None):
  return f[0](2, f[1]) if f else 2
def three(f=None):
  return f[0](3, f[1]) if f else 3
def four(f=None):
  return f[0](4, f[1]) if f else 4
def five(f=None):
  return f[0](5, f[1]) if f else 5
def six(f=None):
  return f[0](6, f[1]) if f else 6
def seven(f=None):
  return f[0](7, f[1]) if f else 7
def eight(f=None):
  return f[0](8, f[1]) if f else 8
def nine(f=None):
  return f[0](9, f[1]) if f else 9


def plus(r, l=None):
  if l is not None:
    return int(eval(f"{r} + {l}"))
  return plus, r
def minus(r, l=None):
  if l is not None:
    return int(eval(f"{r} - {l}"))
  return minus, r
def times(r, l=None):
  if l is not None:
    return int(eval(f"{r} * {l}"))
  return times, r
def divided_by(r, l=None):
  if l is not None:
    return int(eval(f"{r} / {l}"))
  return divided_by, r

File: blakej/js/calculating-with-functions.js
const runTests = require('./run-tests');

const zero = fn => useNumber(fn, 0);
const one = fn => useNumber(fn, 1);
const two = fn => useNumber(fn, 2);
const three = fn => useNumber(fn, 3);
const four = fn => useNumber(fn, 4);
const five = fn => useNumber(fn, 5);
const six = fn => useNumber(fn, 6);
const seven = fn => useNumber(fn, 7);
const eight = fn => useNumber(fn, 8);
const nine = fn => useNumber(fn, 9);

const plus = x => y => y + x;
const minus = x => y => y - x;
const times = x => y => y * x;
const dividedBy = x => y => Math.floor(y / x);

function useNumber(fn, n) {
  if (fn) {
    return fn(n);
  }

  return n;
}

runTests([
  [seven(times(five())), 35],
  [four(plus(nine())), 13],
  [eight(minus(three())), 5],
  [six(dividedBy(two())), 3],
  [four(dividedBy(six())), 0]
]);